dart/lib/src/default_integrations.dart

```diff
@@ -7,16 +7,19 @@ import 'sentry.dart';
 import 'sentry_options.dart';
 import 'throwable_mechanism.dart';
 
-/// integration that capture errors on the runZonedGuarded error handler
+/// integration that runs other integrations within runZonedGuarded to catch any
+/// errors in Dart code running ‘outside’ the Flutter framework
 class RunZonedGuardedIntegration extends Integration {
-  final AppRunner _appRunner;
+  RunZonedGuardedIntegration(this._integrations);
 
-  RunZonedGuardedIntegration(this._appRunner);
+  final List <Integration>_integrations;
 
   @override
-  FutureOr<void> call(Hub hub, SentryOptions options) {
-    runZonedGuarded(() async {
-      await _appRunner();
+  FutureOr<void> call(Hub hub, SentryOptions options) async {
+    await runZonedGuarded(() {
+      for (final integration in _integrations) {
+         integration(hub, options);
+      }
```

Comment:
integrations should be `awaited`, they need to execute synchronously one after the other

Reply:
Had an issue when running a unit test, will check this again.

Reply:
maybe your test must be `async` and you need to `await` the method call within a test.
I had this problem too (if the same)